### PR TITLE
Fix n flag 

### LINKED ISSUE #317: Manim scene caching does not play well with the -n flag
Here's a scene

```python
from manim import Scene, Integer, ShowCreation, Transform

class TestScene(Scene):
    def construct(self):
        number = Integer(0)
        self.add(number)
        for i in range(10):
            self.play(Transform(number, Integer(i)))
```

### Problem 1

This scene cannot be rendered with `manim -pl test.py -n5,10`. That is, if **the first time you render it**, you try to render only a specific number of animations, it cannot be done. It can be rendered normally without specifying the -n flag: `manim -pl test.py`.

### Problem 2
After successfully rendering with `manim -pl test.py`, I tried again `manim -pl test.py -n5,10`, but the whole scene is being rendered. The -n flag is ignored.

### Comments

Judging by the output, I think this has to do with scene caching (cc @huguesdevimeux). I recommend that while we are working out the kinks of the scene cachinc system, we add a flag `--disable_caching`, so that it can still be used even while the caching system is not fully debugged.

UPDATE: even when using `--disable_caching`, you get the same behavior.

### PR DESCRIPTION
**Pull request by huguesdevimeux**
## List of Changes
- Fixes #317

## Explanation for Changes
I modified the mechanism behind -n flag. Before, manim rendered all the partial movie files and selected only the ones with the index between i and j (where i and j are integreers passed with -n flag, eg `-n 5,6`). 
Now, it skips the entire process if an animation is outside the [i,j] interval and creates partial movies files otherwise.

## Testing Status
Created a test for n flag in `test_scene_rendering/test_cli_flags.py`

### CODE REVIEW

File: manim/config/config_utils.py [hunk 1]
```diff
@@ -147,9 +147,7 @@ def _parse_file_writer_config(config_parser, args):
     }
 
     # For internal use (no CLI flag)
-    fw_config["skip_animations"] = any(
-        [fw_config["save_last_frame"], fw_config["from_animation_number"]]
-    )
+    fw_config["skip_animations"] = fw_config["save_last_frame"]
```
Comment (leotrs) [inline manim/config/config_utils.py:150]: Question: can you explain why this change is necessary?

File: manim/scene/scene.py [hunk 2]
```diff
@@ -85,7 +86,7 @@ def __init__(self, **kwargs):
         self.tear_down()
         # We have to reset these settings in case of multiple renders.
         file_writer_config["skip_animations"] = False
```
Comment (leotrs) [inline manim/scene/scene.py:88]: Issue: It should be reset to `self.original_skipping_status`, not to `False`.

File: manim/scene/scene.py [hunk 3]
```diff
@@ -873,7 +880,6 @@ def handle_play_like_call(func):
         """
 
         def wrapper(self, *args, **kwargs):
-            self.update_skipping_status()
```
Comment (leotrs) [inline manim/scene/scene.py:876]: Can you please explain why it is not necessary here to update the skipping status?

File: tests/utils/logging_tester.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
+from functools import wraps
```
Comment (leotrs) [inline tests/utils/logging_tester.py:1]: This file looks like it doesn't belong to this PR, no?

**Review by leotrs — COMMENTED** [review]
Thanks! Question tho: you mention you added a test to test_cli_flags, but I don't see that file in the diff.

**Comment by huguesdevimeux** [comment]
> Thanks! Question tho: you mention you added a test to test_cli_flags, but I don't see that file in the diff.

@leotrs THis is weird. Git is doing crazy things. Nvm, I fixed that :D

File: manim/scene/scene.py (hunk 3 above)
Comment (huguesdevimeux) [reply manim/scene/scene.py:876]: Because now it will be done before, in `handle_caching_play` or `handle_caching_wait` !

File: manim/config/config_utils.py (hunk 1 above)
Comment (huguesdevimeux) [reply manim/config/config_utils.py:150]: Because, setting `skip_animation` to True here because of `from_animation_number` here does not make any sense, as skip_animation is set in consequence of this flag already in `update_skipping_status`, 

Furthermore, (this is a particular case) when one is trying to start from the Animation 0 with `-n 0,2` manim will skip the first animation as `skip_animations` would have been set to True here.

File: manim/scene/scene.py (hunk 2 above)
Comment (huguesdevimeux) [reply manim/scene/scene.py:88]: Hmm fair point, I disagree. scene.original_skipping_status value is set to the one of `file_writer_config[skip_animations]` at the beginning of `__init__`, so it will be set to False in the next scene instantiation.

My concern is when scene.original_skipping_status has been changed for a reason or another. If it is the case and we do your suggestion, then `scene.original_skipping_status` will never be set to False again, for the reason above.

**Review by leotrs — APPROVED** [review]

File: tests/test_scene_rendering/test_cli_flags.py [hunk 5]
```diff
@@ -44,3 +44,24 @@ def test_basic_scene_l_flag(tmp_path, manim_cfg_file, simple_scenes_path):
...
+        "-m",
+        "manim",
+        simple_scenes_path,
+        scene_name,
+        "-n 3,6",
+        "--media_dir",
+        str(tmp_path),
+    ]
+    _, err, exit_code = capture(command)
+    assert exit_code == 0, err
```
Comment (Aathish04) [inline tests/test_scene_rendering/test_cli_flags.py:67]:
```suggestion
    assert exit_code == 0, err

```

Just added a newline at the end.

File: tests/test_scene_rendering/simple_scenes.py [hunk 6]
```diff
@@ -6,3 +6,12 @@ def construct(self):
...
         self.play(Transform(square, circle))
+
+
+class SceneWithMultipleCalls(Scene):
+    def construct(self):
+        number = Integer(0)
+        self.add(number)
+        for i in range(10):
+            number.become(Integer(i))
+            self.play(Animation(number))
```
Comment (Aathish04) [inline tests/test_scene_rendering/simple_scenes.py:17]:
```suggestion
            self.play(Animation(number))

```

Just added a newline at the end.

**Review by Aathish04 — APPROVED** [review]
Approved!

**Comment by huguesdevimeux** [comment]
> Approved!

Thanks !
